src/commands/lightning/preview/app.ts

```diff
@@ -116,22 +193,104 @@ export default class LightningPreviewApp extends SfCommand<void> {
     //
     // Here we've chosen the second approach
     const idx = this.argv.findIndex((item) => item.toLowerCase() === '-o' || item.toLowerCase() === '--target-org');
+    let targetOrg: string | undefined;
     if (idx >= 0 && idx < this.argv.length - 1) {
-      orgOpenCommandArgs.push('--target-org', this.argv[idx + 1]);
+      targetOrg = this.argv[idx + 1];
     }
 
-    await this.config.runCommand('org:open', orgOpenCommandArgs);
+    const launchArguments = PreviewUtils.generateDesktopPreviewLaunchArguments(ldpServerUrl, appId, targetOrg);
+
+    await this.config.runCommand('org:open', launchArguments);
   }
 
-  // eslint-disable-next-line class-methods-use-this, @typescript-eslint/no-unused-vars
-  private async iosPreview(logger: Logger, flags: LightningPreviewAppFlags): Promise<void> {
-    logger.info('Preview on iOS Not Implemented Yet');
-    return Promise.reject(new Error('Preview on iOS Not Implemented Yet'));
+  private async mobilePreview(
+    platform: Platform.ios | Platform.android,
+    ldpServerUrl: string,
+    appName?: string,
+    appId?: string,
+    deviceId?: string,
+    logger?: Logger
+  ): Promise<void> {
+    // 1. Verify that user environment is set up for mobile (i.e. has right tooling)
+    await this.verifyMobileRequirements(platform, logger);
+
+    // 2. Fetch the target device
+    const device = await PreviewUtils.getMobileDevice(platform, deviceId, logger);
+    if (!device) {
+      return Promise.reject(new Error(messages.getMessage('error.device.notfound', [deviceId ?? ''])));
+    }
+
+    // 3. Boot the device if not already booted
+    const resolvedDeviceId = platform === Platform.ios ? (device as IOSSimulatorDevice).udid : device.name;
+    const emulatorPort = await PreviewUtils.bootMobileDevice(platform, resolvedDeviceId, logger);
+
+    // 4. Generate self-signed certificate and wait for user to install it
+    const certFilePath = PreviewUtils.generateSelfSignedCert(platform, '~/Desktop/cert');
```

Comment:
Once the WI for creating `lwr.config.json` is done, this will be updated to save the cert in the same location as the config file